### PR TITLE
Manifest v2 reference

### PR DESCRIPTION
**Pull request by itowlson**
This is just the reference.  We'll need to update other pages too but I wanted to get this out there.

@lann This is written from a single-trigger-type POV because that's what we have today - I know the format is more flexible but don't want to imply things that won't yet work.  Hopefully the rest of it is true though!

@tpmccallum A couple of ponderables for you (on top of wanting your feedback on the content!):

* I'm wondering if it would be good to have a "note for people familiar with Spin 1.x" style, distinct from our normal aside style, perhaps with an icon or something.  That's not intended to replace "if you're familiar with..." wording, but to draw the eye for folks who are migrating, and allow the eye to slide over for folks who are not!
* For this page specifically, I've put the v2 reference and v1 reference both on the one page.  The v1 reference is still relevant enough that I don't want to throw it out, but it does take up a lot of space and make the page structure harder to pick out at a glance.  (Yes, even harder than it was.)  Should we split this into two pages?  How would we want to present that in the TOC - same level in the Reference section, or a new "Spinning It Old Skool" section, or something?

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

✅ content/spin/v2/manifest-reference.md
✅ content/spin/v2/manifest-reference-v1.md

- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-10-30 at 07 49 57
Screenshot 2023-10-30 at 07 49 45

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors

Screenshot 2023-10-30 at 07 52 06

- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by tpmccallum** [comment]
Hi @itowlson 
Thanks, I am leaning into having V1 and V2 pages. This is a lot of content for one page. Also being mindful of folks on smaller devices and phones, trying to read this page :grimacing: 

Would you consider the separate page option? The 2-page option will allow us to remove the 30-odd instances of the words "Version 1" from that page (and its ToC). 
We could then create a separate menu item under the top-level `Reference` heading (and name the page and title to match)
- `Spin V1 Application Manifest`
- `Spin V2 Application Manifest`
A massive thanks for editing this page. A big undertaking that is greatly appreciated.
I am happy with 2-page approach.

**Comment by itowlson** [comment]
I would _totally_ consider the separate page option!  Agree that v1 and v2 manifest side by side in the TOC are easiest for now.  I'll get that under way.

**Comment by itowlson** [comment]
@tpmccallum Split done

File: content/spin/manifest-reference.md [hunk 1]
```diff
@@ -27,42 +32,47 @@ The manifest is a TOML file, and follows standard TOML syntax.  See the [TOML do
...
+
+> If you're familiar with manifest version 1, note that the way trigger parameters map to components - for example, which component handles a particular HTTP route - is now defined on the _trigger_, not on the component. In the version 2 manifest, a `component` section specifies _only_ the Wasm module and the resources it needs.
+
+## The `application` Table
+
+| Name                    | Required?  | Type        | Value    | Example   |
+|-------------------------|------------|-------------|----------|-----------|
 | `name`                  | Required   | String      | The name of the application. This may be any string of alphanumeric characters, hyphens and underscores. | `"hello-world"` |
 | `version`               | Required   | String      | The version of the application. The must be a string of the form `major.minor.patch`, where each element is a number. | `"1.0.5"` |
```
Comment (lann) [inline content/spin/manifest-reference.md:48]: This is no longer mandatory for `spin up` at least.

```suggestion
| `version`               | Optional   | String      | The version of the application. The must be a string of the form `major.minor.patch`, where each element is a number. | `"1.0.5"` |
```

File: content/spin/manifest-reference.md [hunk 2]
```diff
@@ -27,42 +32,47 @@ The manifest is a TOML file, and follows standard TOML syntax.  See the [TOML do
...
+
+> If you're familiar with manifest version 1, note that this is now optional. The trigger type to which the application responds is determined by the top-level `trigger` mappings; you only need to provide an `application.trigger` if you want to override the default global settings for the trigger type.
 
-### The `trigger` Table for HTTP Applications
+### The `application.trigger.http` Table
 
 | Name                    | Required?  | Type        | Value    | Example   |
 |-------------------------|------------|-------------|----------|-----------|
-| `type`                  | Required   | String      | Always `"http"` for HTTP applications. | `"http"` |
 | `base`                  | Required   | String      | The base path of the application. All component routes are relative to this. It allows multiple applications to be mounted under the same host. | `"/"` |
```
Comment (lann) [inline content/spin/manifest-reference.md:65]:
```suggestion
| `base`                  | Optional   | String      | The base path of the application. All component routes are relative to this. It allows multiple applications to be mounted under the same host. | `"/"` |
```

File: content/spin/manifest-reference.md [hunk 3]
```diff
@@ -119,13 +121,38 @@ Each table in the `component` array contains the following fields:
...
+### Additional Fields for `trigger.redis` Tables
 
 | Name                    | Required?  | Type        | Value    | Example   |
 |-------------------------|------------|-------------|----------|-----------|
 | `channel`               | Required   | String      | The Redis channel which this component handles. Messages on this channel will cause the component to execute. | `"purchases"` |
 
-### The `component.build` Table
+## The `component` Table
+
+The keys of the `component` table are user-defined, and are usually written as part of the table syntax e.g. `[component.my-component]`.  The value of each key is another table with the fields shown in the following table.
```
Comment (lann) [inline content/spin/manifest-reference.md:132]: > The keys of the `component` table are user-defined,

This threw me for a loop for a minute thinking that it was referring to the inner table keys and would probably be even more confusing to anyone less familiar with TOML terminology. Perhaps clarify something like:

"The keys of the `component` table (e.g. `my-component` in `[component.my-component]`)"

File: content/spin/manifest-reference.md (hunk 2 above)
Comment (itowlson) [reply content/spin/manifest-reference.md:65]: Oops!  Thanks

File: content/spin/manifest-reference.md (hunk 3 above)
Comment (itowlson) [reply content/spin/manifest-reference.md]: Good catch - thank you.  I've updated as you suggest, and the resulting text is:

> The keys of the `component` table - such as the `my-component` in `[component.my-component]` - are user-defined, and are usually written as part of the table syntax e.g. `[component.my-component]`.

It's clear, but do you think it's okay that it repeats the "[component.my-component]" thing?  I can play around with the wording if not.

Comment (lann) [reply content/spin/manifest-reference.md]: More clear but yeah a bit awkward.

Also I just noticed:

> table with the fields shown in the following table.

which might cause table-madness. Maybe "table with the following fields:"

Comment (itowlson) [reply content/spin/manifest-reference.md]: Huh, that is indeed a lot of tables.  Thanks!

**Comment by tpmccallum** [comment]
Thinking a bit more about this PR, what are your thoughts around keeping both the V1 and V2 pages alive on the `latest` equivalent? I think having both pages present and in the menu (on the frozen content that will be copied off into the path of `/v1` and also on the current docs) will give us a chance to explain that there is a `v1` and a `v2` and they are a bit different (and the application manifest is how you configure an application etc. etc.). Long story short, this PR can go ahead as planned and remain a part of the current content moving forward. Does that sound good?

**Comment by itowlson** [comment]
Yes, this is one of the cases where I believe we should (at least initially) retain v1 content on latest.  The v1 manifest works fine with v2, and users may be maintaining and updating v1 applications.

**Comment by tpmccallum** [comment]
So essentially we would:
- merge this PR,
- then copy off the frozen 1.0 flavoured content,
- then update current content to be 2.0 flavoured.

**Comment by itowlson** [comment]
I am a bit confused now.  We'd probably want to snapshot the v1 content _before_ merging this PR, wouldn't we?

(Also we don't want to merge this PR until we release v2.  Unless we have the v2 content merged, but only surface v1 until we are ready to swap v2 in as latest?  Maybe we need to chat.  MAYBE WE NEED A WHITEBOARD.)

**Comment by itowlson** [comment]
This is going to need some sidebar surgery too.  I backed the TOC change out from here because the v1 reference link would have been bad, but it will need to be reinstated once we figure out the strategy.

**Comment by itowlson** [comment]
bump

**Review by tpmccallum — APPROVED** [review]
LGTM
I will make a note of the sidebar surgery required as part of the flip-switch procedure.

**Pull request merged**